src/vm/database/clarity_db.rs

```diff
@@ -680,6 +716,41 @@ impl<'a> ClarityDatabase<'a> {
             .expect("Failed to get block data.")
     }
 
+    /// Gets the `StacksBlockId` for the *current* chain tip Stacks block height, converts this
+    /// `StacksBlockId` to a `SortitionId` and then gets the `BurnchainHeaderHash` from the
+    /// `BurnStateDB` using this `SortitionId`.
+    ///
+    /// Returns Some() if `0 <= height < current_burnblock_height`, and None otherwise.
+    pub fn get_burnchain_block_header_hash_for_burnchain_height(
+        &mut self,
+        burnchain_block_height: u32,
+    ) -> Option<BurnchainHeaderHash> {
+        let current_stacks_height = self.get_current_block_height();
+        let id_bhh = self.get_index_block_header_hash(current_stacks_height);
+
+        // Is this block in the database?
+        // Should this return an error/panic if it fails?
+        let consensus = match self.headers_db.get_consensus_hash_for_block(&id_bhh) {
+            None => {
+                return None;
+            }
+            Some(consensus) => consensus,
+        };
+
+        // Should this return an error/panic if it fails?
+        let snapshot = match self
+            .burn_state_db
+            .get_block_snapshot_from_consensus_hash(&consensus)
+        {
+            None => {
+                return None;
+            }
+            Some(snapshot) => snapshot,
+        };
+        self.burn_state_db
+            .get_burn_header_hash(burnchain_block_height, &snapshot.sortition_id)
```

Comment:
As part of this PR, can you add a rustdoc to the interface method `get_burn_header_hash` that indicates that it returns the burnchain header has for the given burn block height, as queried from the given Sortition ID?